core/config/engine.h

```diff
@@ -101,6 +101,12 @@ class Engine {
 
 	bool freeze_time_scale = false;
 
+	double time_scale_base = 1.0;
+	double time_scale_override = 1.0;
```

Comment:
I would move these next to `_time_scale` and rename them to `_time_scale_game` and `_time_scale_user`. Perhaps there's better names, but to me "override" implies it _replaces_ the original time scale, not modulate it. 
The setters and getters should also be renamed accordingly.